pkg/api/v1/clients/kube/cache/cache.go

```diff
@@ -136,114 +155,144 @@ func NewKubeCoreCacheWithOptions(ctx context.Context, client kubernetes.Interfac
-				typeCtx = ctxWithTags
-			}
-			// ConfigMap
-			watch := client.CoreV1().ConfigMaps(nsToWatch).Watch
-			list := func(options metav1.ListOptions) (runtime.Object, error) {
-				return client.CoreV1().ConfigMaps(nsToWatch).List(ctx, options)
-			}
-			informer := skkube.NewSharedInformer(typeCtx, resyncDuration, &v1.ConfigMap{}, list, watch)
-			informers = append(informers, informer)
-			lister := kubelisters.NewConfigMapLister(informer.GetIndexer())
-			configMaps[nsToWatch] = lister
-		}
-		{
-			var typeCtx = nsCtx
-			if ctxWithTags, err := tag.New(nsCtx, tag.Insert(skkube.KeyKind, "Secrets")); err == nil {
-				typeCtx = ctxWithTags
-			}
-			// Secrets
-			watch := client.CoreV1().Secrets(nsToWatch).Watch
-			list := func(options metav1.ListOptions) (runtime.Object, error) {
-				return client.CoreV1().Secrets(nsToWatch).List(ctx, options)
-			}
-			informer := skkube.NewSharedInformer(typeCtx, resyncDuration, &v1.Secret{}, list, watch)
-			informers = append(informers, informer)
-			lister := kubelisters.NewSecretLister(informer.GetIndexer())
-			secrets[nsToWatch] = lister
-		}
-
-	}
-
-	var namespaceLister kubelisters.NamespaceLister
-	if len(namesapcesToWatch) == 1 && namesapcesToWatch[0] == metav1.NamespaceAll {
-
-		// Pods
-		watch := client.CoreV1().Namespaces().Watch
-		list := func(options metav1.ListOptions) (runtime.Object, error) {
-			return client.CoreV1().Namespaces().List(ctx, options)
-		}
-		nsCtx := ctx
-		if ctxWithTags, err := tag.New(nsCtx, tag.Insert(skkube.KeyNamespaceKind, skkube.NotEmptyValue(metav1.NamespaceAll)), tag.Insert(skkube.KeyKind, "Namespaces")); err == nil {
-			nsCtx = ctxWithTags
-		}
-		informer := skkube.NewSharedInformer(nsCtx, resyncDuration, &v1.Namespace{}, list, watch)
-		informers = append(informers, informer)
-		namespaceLister = kubelisters.NewNamespaceLister(informer.GetIndexer())
-	}
-
 	k := &kubeCoreCaches{
 		podListers:       pods,
 		serviceListers:   services,
 		configMapListers: configMaps,
 		secretListers:    secrets,
-		namespaceLister:  namespaceLister,
+		client:           client,
+		ctx:              ctx,
+		resyncDuration:   resyncDuration,
+		informers:        informers,
+	}
+
+	for _, nsToWatch := range namesapcesToWatch {
+		k.addNewNamespace(nsToWatch)
 	}
 
-	kubeController := controller.NewController("kube-plugin-controller",
-		controller.NewLockingSyncHandler(k.updatedOccured), informers...,
+	// TODO allows add in the namespaceLister, all the time, or make it an option.
+	// TODO-JAKE the namespace lister is used for something in Gloo, we need to make sure that it is going to
+	// dealt with.  It might have side affects that I am unaware of, so please go and investigate that.
+
+	// TODO-JAKE please look into the Reconciler as well, those might need to updated too. I am not sure though, as they work differently.
+	if len(namesapcesToWatch) == 1 && namesapcesToWatch[0] == metav1.NamespaceAll {
+		k.addNamespaceLister()
+	}
```

Comment:
going to change this so that it will always create the namespace lister no matter what.
Or we could change this to allow for a flag to set the option.

Reply:
this was changed to a flag, because it would allow the client to choose.  The reason was because Gloo needed the option, based off whether it was using RBAC `namespaced:true`